packages/core/src/components/tag-input/tagInput.tsx

```diff
@@ -414,4 +415,15 @@ export class TagInput extends AbstractPureComponent<ITagInputProps, ITagInputSta
     private isValidIndex(index: number) {
         return index !== NONE && index < this.props.values.length;
     }
+
+    private handlePaste = (event: React.ClipboardEvent<HTMLInputElement>) => {
+        // we don't want it to show up in the input since we're directly adding the terms
+        event.preventDefault();
+        const { onChange, values } = this.props;
+        const clipboardText = event.clipboardData.getData("text");
+        const newValues = [].concat(...clipboardText.split(/\n|\r/).map(s => this.getValues(s)));
+        if (Utils.isFunction(onChange)) {
+            onChange([...values, ...newValues]);
```

Comment:
`Utils.safeInvoke()` instead of `if`. see other uses throughout this file.

Reply:
also `Utils.safeInvoke(this.inputProps.onPaste)`

Reply:
Ack - was copying from addTag, will fix it

Reply:
ah heh that one has a good reason to use `isFunction` directly